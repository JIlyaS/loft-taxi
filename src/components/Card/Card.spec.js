// import { mount } from 'enzyme';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';

import {Card} from './';

// const defaultProps = {
//   date: null,
//   numberCard: null,
// };

// const TestCard = props => <Card {...defaultProps} {...props} />;

describe('Card', () => {

  it('renders correctly snapshot', () => {
    const tree = renderer.create(<Card />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('render correct', () => {
    const { queryByTestId } = render(<Card />);
    expect(queryByTestId('card-date')).toBeInTheDocument();
    expect(queryByTestId('card-number')).toBeInTheDocument();
  });
  
  // it('render correct', () => {
  //   const { queryByTestId } = render(<Map />);
  //   expect(queryByTestId('map')).toBeInTheDocument();
  // });

  // it("render card correctly with null props", () => {
  //   const props = {
  //     date: null,
  //     numberCard: null,
  //   };
  //   const CardComponent = mount(<TestCard {...props} />);
  //   expect(CardComponent.prop("date")).toEqual(null);
  //   expect(CardComponent.prop("numberCard")).toEqual(null);
  // });


});