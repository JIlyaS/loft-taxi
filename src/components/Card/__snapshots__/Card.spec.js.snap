// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders correctly snapshot 1`] = `
<div
  className="card"
>
  <div
    className="MuiPaper-root MuiCard-root card__ui MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      className="MuiCardContent-root card__content"
    >
      <div
        className="card__content-top"
      >
        <img
          alt="Иконка карточки"
          className="card__icon-road"
          height="34px"
          src="card-road.svg"
          width="34px"
        />
        <span
          className="card__date"
          data-testid="card-date"
        >
          06/21
        </span>
      </div>
      <div
        className="card__content-main"
      >
        <p
          className="card__number"
          data-testid="card-number"
        >
          
        </p>
      </div>
      <div
        className="card__content-bottom"
      >
        <img
          alt="Иконка карточки"
          className="card__icon-card"
          height="26px"
          src="card-icon.svg"
          width="29px"
        />
        <span
          className="card__icon-ellipses"
        />
      </div>
    </div>
  </div>
</div>
`;
